core/src/avm2/globals/int.rs

```diff
@@ -118,140 +130,74 @@ fn class_init<'gc>(
-                .coerce_to_u32(activation)? as usize;
-
-            if digits > 20 {
-                return Err("toFixed can only print with 0 through 20 digits.".into());
-            }
-
-            return Ok(AvmString::new_utf8(
-                activation.context.gc_context,
-                format!("{0:.1$}", number as f64, digits),
-            )
-            .into());
-        }
-    }
-
-    Err("int.prototype.toFixed has been called on an incompatible object".into())
-}
+use crate::avm2::globals::number::to_fixed;
 
 /// Implements `int.toPrecision`
-fn to_precision<'gc>(
-    activation: &mut Activation<'_, 'gc>,
-    this: Object<'gc>,
-    args: &[Value<'gc>],
-) -> Result<Value<'gc>, Error<'gc>> {
-    if let Some(this) = this.as_primitive() {
-        if let Value::Integer(number) = *this {
-            let wanted_digits = args
-                .get(0)
-                .cloned()
-                .unwrap_or(Value::Integer(0))
-                .coerce_to_u32(activation)? as usize;
-
-            if wanted_digits < 1 || wanted_digits > 21 {
-                return Err(Error::AvmError(range_error(activation, "Error #1002: Number.toPrecision has a range of 1 to 21. Number.toFixed and Number.toExponential have a range of 0 to 20. Specified value is not within expected range.", 1002)?));
-            }
-
-            return Ok(print_with_precision(activation, number as f64, wanted_digits)?.into());
-        }
-    }
-
-    Err("int.prototype.toPrecision has been called on an incompatible object".into())
-}
+use crate::avm2::globals::number::to_precision;
 
 /// Implements `int.toString`
 fn to_string<'gc>(
     activation: &mut Activation<'_, 'gc>,
     this: Object<'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    if let Some(this) = this.as_primitive() {
-        if let Value::Integer(number) = *this {
-            let radix = args
-                .get(0)
-                .cloned()
-                .unwrap_or(Value::Integer(10))
-                .coerce_to_u32(activation)? as usize;
-
-            if radix < 2 || radix > 36 {
-                return Err(Error::AvmError(range_error(
-                    activation,
-                    &format!(
-                        "Error #1003: The radix argument must be between 2 and 36; got {radix}."
-                    ),
-                    1003,
-                )?));
-            }
+    let int_proto = activation.avm2().classes().int.prototype();
+    if Object::ptr_eq(int_proto, this) {
+        return Ok("0".into());
+    }
 
-            return Ok(print_with_radix(activation, number as f64, radix)?.into());
+    let number = if let Some(this) = this.as_primitive() {
+        match *this {
+            Value::Integer(o) => o,
+            _ => return Err(make_error_1004(activation, "int.prototype.toString")),
         }
+    } else {
+        return Err(make_error_1004(activation, "int.prototype.toString"));
+    };
+
+    let radix = args
```

Comment:
technically `toString` is also shared with `Number`, right? The only difference being in prototype on `to(Locale)String` checking the proto and throwing different errors. But if you don't think the after-type-checking part can be easily unified, feel free to ignore this, no pressure. Maybe just a comment?

Reply:
I think it's better to keep it this way- the errors are different, and `(u)int` have an additional check in those methods- if `this == prototype`, return "0".

Reply:
I think it's better to keep it this way; converting it would be more annoying. Additionally, we're getting close to having the primitive classes written in AS, too, so it will probably be removed soon-ish.